### ISSUE TITLE
QuickEditMode Issue with console

### ISSUE THREAD

**Issue by MrPogostyc** [mcserversoft/IssueTracker#9]
*I will be contacting the discord as soon as I am home, but for now, I want to document this issue on the Github. If this is against the Issue tracker rules (which to my understanding, this is not), I am more than okay to take this down.*

**Issue**
There is an Issue with the integration of the Command Prompt into the software. This problem I have identified to be caused by Windows 10's "Quick Edit Mode" under the defaults section when you right click the title bar of the Command Prompt. The issue is as follows:
The console "freezes" until prompted with a command by the user. The server works great and isn't actually frozen, however certain actions, such as closing the server, freeze before completing, and the program's kill interval is the only way I have found to get the server to close in the background without requiring me to type a one letter input into the console. I believe this is an issue for not just automation, but also backups. I have the scheduler set up to back up, but it hasn't been able to complete the job since the server freezes at "Server closing..." and requires human input to finish the task. 

This goes without saying, but just for better communication, My issue here is that this "bug" or whatever, defeats the purpose of automated scheduled tasks, such as the scheduled restart, as it cant do it whilst I'm away, as it requires human input to close properly.

The issue on its own, outside of MCSS is fixed by turning off this "QuickEditMode" in the command prompt. However, the integrated console on MCSS does not include this feature, which defeats the option to have total automated restarts.

**Describe the solution you'd like**
I'd like for an update to the software that includes the ability to check off this "QuickEditMode" under the default section as would appear in the command prompt normally if I had just normally ran my batch file. 

**Describe alternatives you've considered**
Of course, I could just run the batch file normally, but that would defeat the entire purpose of using this Software. I want to be able to schedule tasks and have remote control so when I away from home, So this isn't an option in this given context. 

**Additional context**
![image](https://user-images.githubusercontent.com/59030023/213249360-0804287b-72ab-40c5-afc2-b2b9837e5c82.png)
![image](https://user-images.githubusercontent.com/59030023/213249429-188f1dc6-8eaa-4f1f-a37d-5162012612a5.png)
![image](https://user-images.githubusercontent.com/59030023/213249496-190e74c1-4592-4acd-ace2-f71d7eabedcc.png)
Above is the described process of disabling QuickEditMode inside of Command Prompt. 

Below is a thread describing the problem and identifying the solution:
https://www.minecraftforum.net/forums/support/server-support-and/3017626-a-solution-to-minecraft-server-console-freezing

**Comment by PouletteMC** [comment]
Do you reckon you could dumb that down for me ?

No rush though, take your time top get home

**Comment by Fiahblade** [comment]
@MrPogostyc Thank you for taking the time for reporting the issue.
While I haven't heard of this issue before, it seems to be pretty old: 
https://stackoverflow.com/questions/30418886/how-and-why-does-quickedit-mode-in-command-prompt-freeze-applications

So to recap, commands don't get executed?

**Comment by MrPogostyc** [comment]
Allow me to rephrase to the best of my ability. The issue here is, frequently the server command prompt freezes, and doesn't display lines or progress the stop server process until I type in anything. It always gets stuck at "Closing Server..." and does not actually close the server all the way UNLESS, I send in a command. It doesn't matter what command I send, it just needs to be prompted to finish closing and print the line "Press any key to continue...". Only after the server has closed can MCSS start the server back up again.

***Breakdown of what happens:***
The scheduler sends the command to stop the server it at 8AM, but it gets stuck at "Closing Server..." and can't continue UNTIL I come in and type in a command. Once I DO type in a command, it closes properly and starts back up properly. 

***Why is this an issue?***
Because currently, I have it set up, to restart the server every day at 8 AM while I am heading off to school and not currently playing the server. This was deliberate on my part so I can have it restart during an hour that I or most people aren't online. That was the whole appeal of this program, so that I could automate processes like restarts without needing to do anything manually and during an hour of the day that I was not at the computer. However, the issue described ***makes*** the process manual, which defeats the whole purpose of having the program in the first place. The server doesn't start back up when its supposed to, and I have to go in every single time it gets frozen, and type in something to continue it. 

I cannot do that when I am away or busy.

With this issue present, It renders this program useless to me, because if I have to be present at the computer and manually get the server started again, might as well just run the batch file myself and not add an extra step.

Its important to note that this is an issue that was effecting me before I started using MCSS and isn't a direct problem of MCSS in it of itself, but rather, the command prompt has a feature Called "QuickEditMode" That creates this issue. 

According to my experience and the[ Minecraft Forum thread](https://www.minecraftforum.net/forums/support/server-support-and/3017626-a-solution-to-minecraft-server-console-freezing) I linked earlier, turning off the QuickEditMode in the Command Prompt fixes this issue, however, the option to turn off QuickEditMode in the Integrated command prompt in MCSS does not exist. 

***My suggestion is as follows:***
Add a fix in an update that turns off, or at least gives the option to turn off "QuickEditMode" in the integrated Command Prompt inside of MCSS so that I can continue to use this software for my intended usage.

***Other things to note about the Issue:***
For some reason, the Server still works perfectly fine when this happens while the server is running, I just have to prompt it, then all the lines it should have shown up to that point appear out of no where, which isn't a big deal as the server still works fine, I just have to prompt it if I want to see the real time log up that point, until the server freezes again. To reiterate, as far as I've experienced, the only process that gets "frozen" is the process of closing the server. When people are online, commands still work and get logged. just if I want to see those logs, I have to type something in the command prompt, and then it will shoot out EVERY line that should have been printed since freezing. 

***PASTED LOG:***
If I run the command "/stop", click on the restart, or click on the stop button IN MCSS, the following occurs, which to my understanding, is normal, but comes to an abrupt halt once it displays "Closing Server..." and requires me to send in a command to continue the task of closing the server.

[11:11:41 INFO]: Stopping the server
[11:11:41 INFO]: Stopping server
[11:11:41 INFO]: [CoreProtect] Disabling CoreProtect v21.3
[11:11:41 INFO]: [CoreProtect] Finishing up data logging. Please wait...
[11:11:43 INFO]: [CoreProtect] Success! Disabled CoreProtect v21.3
[11:11:43 INFO]: [ServerlistMOTD] Disabling ServerlistMOTD vX-2021-11-23
[11:11:43 INFO]: [dynmap-structures] Disabling dynmap-structures v2.6.0
[11:11:43 INFO]: [EssentialsChat] Disabling EssentialsChat v2.19.7
[11:11:43 INFO]: [Multiverse-Inventories] Disabling Multiverse-Inventories v4.2.3-b523
[11:11:43 INFO]: [Essentials] Disabling Essentials v2.19.7
[11:11:43 INFO]: [Vault] [Economy] Essentials Economy unhooked.
[11:11:43 INFO]: [Chunky] Disabling Chunky v1.3.52
[11:11:43 INFO]: [WorldEdit] Disabling WorldEdit v7.2.13+46576cc
[11:11:43 INFO]: Unregistering com.sk89q.worldedit.bukkit.BukkitServerInterface from WorldEdit
[11:11:43 INFO]: [dynmap] Disabling dynmap v3.4-828
[11:11:43 INFO]: [dynmap] Unloaded 11 components.
[11:11:43 INFO]: [dynmap] Stopping map renderer...
[11:11:44 INFO]: [dynmap] Disabled
[11:11:44 INFO]: [Multiverse-Core] Disabling Multiverse-Core v4.3.1-b861
[11:11:44 INFO]: [Vault] Disabling Vault v1.7.3-b131
[11:11:44 INFO]: [Geyser-Spigot] Disabling Geyser-Spigot v2.1.0-SNAPSHOT
[11:11:44 INFO]: [Geyser-Spigot] Shutting down Geyser.
[11:11:44 INFO]: [Geyser-Spigot] Geyser shutdown successfully.
[11:11:44 INFO]: [voicechat] Disabling voicechat v1.19.2-2.3.24
[11:11:44 INFO]: [LuckPerms] Disabling LuckPerms v5.4.58
[11:11:44 INFO]: [LuckPerms] Starting shutdown process...
[11:11:44 INFO]: [LuckPerms] Closing storage...
[11:11:44 INFO]: [Geyser-Spigot] The skin uploader has been closed
[11:11:44 INFO]: [LuckPerms] Goodbye!
[11:11:44 INFO]: [ChairStairs] Disabling ChairStairs v1.1
[11:11:44 INFO]: [ChairStairs] has been disabled.
[11:11:44 INFO]: [floodgate] Disabling floodgate v2.2.0-SNAPSHOT (b74-4f36112)
[11:11:44 INFO]: [HolographicDisplays] Disabling HolographicDisplays v3.0.0
[11:11:44 INFO]: [ViaVersion] Disabling ViaVersion v4.5.2-SNAPSHOT
[11:11:44 INFO]: [ViaVersion] ViaVersion is disabling, if this is a reload and you experience issues consider rebooting.
[11:11:44 INFO]: [NotTooExpensive] Disabling NotTooExpensive v1.3.4
[11:11:44 INFO]: [AntiCreeper] Disabling AntiCreeper v5.1
[11:11:44 INFO]: [ChatColor2] Disabling ChatColor2 v1.12.4
[11:11:44 INFO]: [ChatColor] ChatColor 2 Version 1.12.4 has been disabled.
[11:11:44 INFO]: [OldCombatMechanics] Disabling OldCombatMechanics v1.11.0
[11:11:44 INFO]: [OldCombatMechanics] OldCombatMechanics v1.11.0 has been disabled
[11:11:44 INFO]: [ImageMaps] Disabling ImageMaps v1.0.10
[11:11:44 INFO]: Saving players
[11:11:44 INFO]: Saving worlds
[11:11:44 INFO]: Saving chunks for level 'ServerLevel[world]'/minecraft:overworld
[11:11:44 INFO]: [ChunkHolderManager] Waiting 60s for chunk system to halt for world 'world'
[11:11:44 INFO]: [ChunkHolderManager] Halted chunk system for world 'world'
[11:11:44 INFO]: [ChunkHolderManager] Saving all chunkholders for world 'world'
[11:11:44 INFO]: [ChunkHolderManager] Saved 0 block chunks, 74 entity chunks, 0 poi chunks in world 'world' in 0.01s
[11:11:46 INFO]: ThreadedAnvilChunkStorage (world): All chunks are saved
[11:11:46 INFO]: Saving chunks for level 'ServerLevel[world_nether]'/minecraft:the_nether
[11:11:46 INFO]: [ChunkHolderManager] Waiting 60s for chunk system to halt for world 'world_nether'
[11:11:46 INFO]: [ChunkHolderManager] Halted chunk system for world 'world_nether'
[11:11:46 INFO]: [ChunkHolderManager] Saving all chunkholders for world 'world_nether'
[11:11:46 INFO]: [ChunkHolderManager] Saved 141 block chunks, 27 entity chunks, 0 poi chunks in world 'world_nether' in 0.12s
[11:11:47 INFO]: ThreadedAnvilChunkStorage (DIM-1): All chunks are saved
[11:11:47 INFO]: Saving chunks for level 'ServerLevel[world_the_end]'/minecraft:the_end
[11:11:47 INFO]: [ChunkHolderManager] Waiting 60s for chunk system to halt for world 'world_the_end'
[11:11:47 INFO]: [ChunkHolderManager] Halted chunk system for world 'world_the_end'
[11:11:47 INFO]: [ChunkHolderManager] Saving all chunkholders for world 'world_the_end'
[11:11:47 INFO]: [ChunkHolderManager] Saved 0 block chunks, 69 entity chunks, 0 poi chunks in world 'world_the_end' in 0.01s
[11:11:47 INFO]: ThreadedAnvilChunkStorage (DIM1): All chunks are saved
[11:11:47 INFO]: Saving chunks for level 'ServerLevel[jacobland]'/minecraft:jacobland
[11:11:47 INFO]: [ChunkHolderManager] Waiting 60s for chunk system to halt for world 'jacobland'
[11:11:47 INFO]: [ChunkHolderManager] Halted chunk system for world 'jacobland'
[11:11:47 INFO]: [ChunkHolderManager] Saving all chunkholders for world 'jacobland'
[11:11:47 INFO]: [ChunkHolderManager] Saved 0 block chunks, 76 entity chunks, 0 poi chunks in world 'jacobland' in 0.01s
[11:11:48 INFO]: ThreadedAnvilChunkStorage (jacobland): All chunks are saved
[11:11:48 INFO]: Saving chunks for level 'ServerLevel[gsmp]'/minecraft:gsmp
[11:11:48 INFO]: [ChunkHolderManager] Waiting 60s for chunk system to halt for world 'gsmp'
[11:11:48 INFO]: [ChunkHolderManager] Halted chunk system for world 'gsmp'
[11:11:48 INFO]: [ChunkHolderManager] Saving all chunkholders for world 'gsmp'
[11:11:48 INFO]: [ChunkHolderManager] Saved 0 block chunks, 109 entity chunks, 0 poi chunks in world 'gsmp' in 0.04s
[11:11:50 INFO]: ThreadedAnvilChunkStorage (gsmp): All chunks are saved
[11:11:50 INFO]: ThreadedAnvilChunkStorage: All dimensions are saved
[11:11:50 INFO]: Flushing Chunk IO
[11:11:50 INFO]: Closing Thread Pool
2023-01-18 11:11:50,026 Log4j2-AsyncAppenderEventDispatcher-1-Async WARN Advanced terminal features are not available in this environment
[11:11:50 INFO]: Closing Server

As you can see, it gets stuck at "Closing Server" and will stay this way forever until I send a command, or have the Server inactivity kill counter enabled, which I would rather not have, because it seems to be creating issues with my scheduled backups.

Hopefully you got the gist of it. If I need to, I can recreate the issue on recording and send it in to better express what I'm trying to say, But I'm really hoping I explained it well enough.

**Comment by MrPogostyc** [comment]
Where would I go about asking this question in the Discord??

**Comment by PouletteMC** [comment]
I want to say the suggestions channel

**Comment by Fiahblade** [comment]
Thank you, will take a look at this soon

**Comment by MrPogostyc** [comment]
Thank you! This issue still plagues me to this day lol

**Comment by Fiahblade** [comment]
@MrPogostyc Can you disable the QuickEditMode in the default tab and see if that affects mcss? (might need to restart)
![image](https://user-images.githubusercontent.com/2144509/216321395-24682526-a221-4786-a132-b1760a3d0cac.png)

**Comment by MrPogostyc** [comment]
I tried that before posting, apologies for not communicating that in the original post. It does not work unfortunately in MCSS. It works while running it in a normal Console, but not in MCSS.

**Comment by Fiahblade** [comment]
I did some experimenting and I believe I can't modify the QuickEditMode flag because the cmd window I'm launching is hidden. Would you mind sharing how your bat file looks like?

**Comment by MrPogostyc** [comment]
Not sure what exactly you are looking for in looking at this Bat file. Could you be more specific? Like do you want a screenshot of just a normal server running or something else?

**Comment by MrPogostyc** [comment]
I believe I have discovered the Issue. I have all of my servers start through a bat file, in which at the end of the batch file, @pause was present. Removing this @pause line seems to have fixed the issue. Apologies for adding any unwanted tasks to your free time of being a developer.
